### DISCUSSION TITLE
Can i install both dwm(xorg) and hyprland on to the same machine?

### DISCUSSION THREAD

**Discussion by ogios** [hyprwm/Hyprland#3081] [Q&A]
i'm new to this, and some apps broke on xwayland, so i want to install a backup window manager that is for xorg which is dwm.  
i haven' t done this before, but i remember that i used to saw both x11 and wayland choices on ubuntu.  
is it possible to install dwm the same time i'm using hyprland?

**Comment by kauer3** [comment]
@ogios 

Yes, you can have both installed on your machine, you just cannot use them simultaneously.

> i remember that i used to saw both x11 and wayland choices on ubuntu

That's right, in your display manager (login screen) you should have a way to choose between all desktop environments and window managers you have installed before logging in.

**Comment by HappyGoFishing** [comment]
You might want to create a wrapper script that sets the environment variables for X/Wayland respectively before launching. (Some programs like firefox require certain wayland variables to be set.

**Comment by Thesoul20** [comment]
So, how to config it ？
